### DISCUSSION TITLE
Rectangular selection functionality

### DISCUSSION THREAD

**Discussion by jkissi** [marcomusy/vedo#616] [Q&A] [answered]
Hello Marco!!
I would like to create an option to select a subset of points from my point cloud and show the properties of those points only. I know that there are click and mouse move callback functions from a previous answer you gave. Simply put, what I am after is a "click and drag" selection box, where you click and hold the button and the viewer starts drawing a box, and you release the mouse button and it finishes drawing the box. Then I can know all of the coordinates of the points within the boundary of the box. I've included an image from another program, just in case I am not explaining it very well...

![sel_box__edit](https://user-images.githubusercontent.com/8246894/158437792-3a0e5d13-65df-47fb-b945-f600fbe94ebb.png)

Are there any examples that do something like this that I could have a look at? If not, if you can think of any suggestions, I would, as always, be very grateful

Cheers
j

**Comment by marcomusy** [comment, marked as answer]
Hi in fact this has been on my my todo list for a while... but I always resisted... :)
It would not be so difficult to implement in vedo as VTK has this functionality already.

Maybe thisis something you'd like to play around if it fits your needs:
(right-click start and stop drawing, the press z or Z to cut)
```python
import numpy as np
from vedo import *
from vedo.applications import FreeHandCutPlotter

pts = np.random.randn(10000, 3)
pts = Points(pts).ps(8)
pts.name = "My Gaussian"

cube = Cube().x(2)

plt = FreeHandCutPlotter(pts)
plt.addHoverLegend()
plt.show(pts, Axes(pts), cube) # cube is ignored by the cutter
plt.close()
```

![Screenshot from 2022-03-15 19-16-29](https://user-images.githubusercontent.com/32848391/158445552-22a89e7e-de52-4cc0-80b4-eb4e60f67bac.png)

![Screenshot from 2022-03-15 19-16-43](https://user-images.githubusercontent.com/32848391/158445583-64fdaaf5-4aca-4494-88e2-66ab702a023f.png)

Check out:
`examples/basic/cutFreeHand.py`

..maybe you can customize that one with a box instead of a line.

**Reply by jkissi** [reply]
Ok thanks for the answer. I need a little time to fiddle with this, so I'll get back to you when I have more questions. Thanks again!!
